src/FuncSharp/Collections/IEnumerableExtensions_ReadOnlyList.cs

```diff
@@ -1,100 +1,122 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
+using System.Diagnostics.Contracts;
 
 namespace FuncSharp;
 
 public static partial class IEnumerableExtensions
 {
-    public static bool NonEmpty<T>(this IReadOnlyList<T> source)
+    [Pure]
+    public static bool NonEmpty<T>(this IReadOnlyCollection<T> collection)
     {
-        return source is not null && source.Count > 0;
+        return collection is not null && collection.Count > 0;
```

Comment:
Will this be the same?

```suggestion
        return collection?.Count > 0;
```

Reply:
Yup, it should be 
<img width="518" alt="image" src="https://github.com/MewsSystems/FuncSharp/assets/24482879/6373a41a-3a56-4ac4-b96d-5bce5f5e930e">
<img width="512" alt="image" src="https://github.com/MewsSystems/FuncSharp/assets/24482879/d08dfa84-6e70-4726-bc6f-8e31679eac4f">

Reply:
Based on our 🤙🏻 we'll keep it as it is